pinot-core/src/main/java/org/apache/pinot/core/common/MinionConstants.java

```diff
@@ -199,4 +200,59 @@ public static class UpsertCompactionTask {
      */
     public static final String NUM_SEGMENTS_BATCH_PER_SERVER_REQUEST = "numSegmentsBatchPerServerRequest";
   }
+
+  public static class UpsertCompactMergeTask {
```

Comment:
broad question: say we run both merge compaction and the segment refresh task for a table. How do we ensure that there aren't any race conditions, because it could be that the two tasks end up processing the same set of segments concurrently.

Reply:
Correctness wise we should be fine I guess, but this will still impede or impact the Segment Refresh task. Can you create a separate issue for this with some context on the problem so we could pick it up later?

Reply:
https://github.com/apache/pinot/issues/14633